resources/deploymentconfig.yaml

```diff
@@ -56,9 +56,30 @@ spec:
         - --end-day
         - "7"
         - --server
+        env:
+          - name: SIPPY_DATABASE_DSN
+            valueFrom:
+              secretKeyRef:
+                name: postgres
+                key: dsn
         volumeMounts:
         - mountPath: /data
           name: data
+      - image: quay.io/enterprisedb/postgresql
+        imagePullPolicy: Always
+        name: postgres
+        resources:
+          limits:
+            memory: 2G
+        env:
+          - name: POSTGRES_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: postgres
+                key: password
+        volumeMounts:
+          - mountPath: /var/lib/postgresql/data
+            name: db
```

Comment:
Is postgresql going to be *required* or optional?

Presumably the kube instance is using separate manifests, as I doubt they have DeploymentConfig.

However is this something we want to lean in hard, or do we want to preserve both methods for now?

For this PR am I correct that bigquery will keep the data permanently for now, so if we were to blow away the postgres db, it would all get re-imported next time through? Are we targeting postgres being something that is easy to throwaway and rebuild via the fetchdata script, and if so is that likely permanent or temporary? In my mind it might be nice to start transient and throwaway, and in time possibly it becomes something we need to monitor and backup.

We'll want to carefully balance with the historical requirements for loading a directory of testgrid data. That should probably be replaced by something in the db ultimately though if it becomes permanent data.

Reply:
> Is postgresql going to be required or optional?

I think if we move report data to it other than bigquery, we should just always put it in a Postgres DB.  WDYT? It’s probably using less resources to do that.

> Presumably the kube instance is using separate manifests, as I doubt they have DeploymentConfig.

Yea, I don’t think they use anything in here.  Also updating it is a manual process. David knows the details.

> For this PR am I correct that bigquery will keep the data permanently for now, so if we were to blow away the postgres db, it would all get re-imported next time through? Are we targeting postgres being something that is easy to throwaway and rebuild via the fetchdata script, and if so is that likely permanent or temporary? In my mind it might be nice to start transient and throwaway, and in time possibly it becomes something we need to monitor and backup.

I had imagined the Postgres always being ephemeral. It’ll be rebuilt from scratch if needed.  I’m not entirely sure I want to be in the business of being responsible for backups, etc — it’s nicer if we can re-derive everything from BigQuery or TestGrid data.  I guess if we want to have comments or other stateful data, we’ll have to come up with a solution for that.


> We'll want to carefully balance with the historical requirements for loading a directory of testgrid data. That should probably be replaced by something in the db ultimately though if it becomes permanent data.